tests/Annotations/Command/QueueImportCommandTest.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace tests\eLife\Annotations\Command;
+
+use eLife\Annotations\Command\QueueImportCommand;
+use eLife\ApiClient\HttpClient;
+use eLife\ApiClient\Result\HttpResult;
+use eLife\ApiSdk\ApiSdk;
+use eLife\Bus\Limit\Limit;
+use eLife\Bus\Queue\WatchableQueue;
+use eLife\Logging\Monitoring;
+use GuzzleHttp\Promise\FulfilledPromise;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit_Framework_TestCase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Tester\CommandTester;
+
+/**
+ * @covers \eLife\Annotations\Command\QueueImportCommand
+ */
+class QueueImportCommandTest extends PHPUnit_Framework_TestCase
+{
+    /** @var ApiSdk */
+    private $apiSdk;
+    /** @var Application */
+    private $application;
+    /** @var QueueImportCommand */
+    private $command;
+    /** @var CommandTester */
+    private $commandTester;
+    private $httpClient;
+    private $limit;
+    private $logger;
+    /** @var Monitoring */
+    private $monitoring;
+    private $queue;
+
+    /**
+     * @before
+     */
+    public function prepareDependencies()
+    {
+        $this->httpClient = $this->getMockBuilder(HttpClient::class)
+            ->setMethods(['send'])
+            ->getMock();
+        $this->apiSdk = new ApiSdk($this->httpClient);
+        $this->application = new Application();
+        $this->limit = $this->createMock(Limit::class);
+        $this->logger = $this->getMockBuilder(LoggerInterface::class)
+            ->setMethods(['emergency', 'alert', 'critical', 'error', 'warning', 'notice', 'info', 'debug', 'log'])
+            ->getMock();
```

Comment:
Better to use something like [BufferingLogger](https://github.com/symfony/symfony/blob/5129c4cf7e294b1a5ea30d6fec6e89b75396dcd2/src/Symfony/Component/Debug/BufferingLogger.php) rather than constructing mocks of external code manually.